### ISSUE TITLE
Lecture01-Information Collection of Accessing GitHub using Command Lines

### ISSUE THREAD

**Issue by ZxjSamantha** [ZxjSamantha/System-Development-Studio#2]
This is work log of information collection of accessing GitHub using command lines.

**Comment by ZxjSamantha** [comment]
Step 1: 
I first searched how to access GitHub from terminal on Google with keyword "github terminal mac" (terminal is written in Chinese). I found a tutorial in Chinese about connecting GitHub from terminal in the following URL.

[Use GitHub to share in the terminal](https://www.jianshu.com/p/899a2d4de88f)

**Comment by ZxjSamantha** [comment]
Step 2:

In this tutorial, the author mentioned ssh key. I used to use ssh key for remote connection. So I change the keyword to "github mac ssh".

I found a GitHub Doc written by GitHub free pro team.

Step 1: Generate a new SSH key and addiing it to the ssh-agent.

[Generating a new SSH key and adding it to the ssh-agent](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)

Step 2: Add the new SSH key to my GitHub account 

[Adding a new SSH key to your GitHub account](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/adding-a-new-ssh-key-to-your-github-account)

Step 3: Test the SSH connection

[Testing your SSH connection](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/testing-your-ssh-connection)

**Comment by ZxjSamantha** [comment]
Step 3:

I searched on Google with keyword "GitHub commit tutorial" in Chinese. Here is the URL of one of the tutorials I found most useful and simple (in Chinese). 

[Simple Tutorial of GitHub](https://www.runoob.com/w3cnote/git-guide.html)

I also found there was a complete free online course provided by Udacity, about How to use git and GitHub. 

[Version Control with Git](https://www.udacity.com/course/version-control-with-git--ud123)

Before creating repositories, the configuration needs to be set with the following commands: 

> $ git config --global user.name "your name"

> $ git config --global user.email "your_email@youremail.com"

Create a new git repository:

> git init

Create a clone of repository on remote server:

> git clone username@host:/path/to/repository

Add the local file to the staging area (this step can be excuted for many files): 

> git add filename
    
Commit the change of code to HEAD of local repository:

> git commit -m "commit description"

Commit the change of code to remote repository:

> git push origin *master or branch*

Create branch of master and switch to this branch:

> git checkout -b branch_name 

Switch back to the master:

> git checkout master

Delete a branch:

> git branch -d branch_name

Update the latest changes to local repository:

> git pull

Merge the branches to current branch (or master) 

> git merge branch

Check the differences before merging:

> git diff source_branch target_branch
